src/module.c

```diff
@@ -3769,209 +3758,131 @@ long long RM_StreamTrimByID(RedisModuleKey *key, int flags, RedisModuleStreamID
-    if (reply->type != REDISMODULE_REPLY_INTEGER) return LLONG_MIN;
-    return reply->val.ll;
+    return callReplyGetLongLong(reply);
+}
+
+/* Return the double value of an double reply. */
+double RM_CallReplyDouble(RedisModuleCallReply *reply) {
+    return callReplyGetDouble(reply);
+}
+
+/* Return the big number value of a big number reply. */
+const char* RM_CallReplyBigNumber(RedisModuleCallReply *reply, size_t* len) {
+    return callReplyGetBigNumber(reply, len);
+}
+
+/* Return the string value of an verbatim string reply. */
+const char* RM_CallReplyVerbatimFormat(RedisModuleCallReply *reply) {
+    return callReplyGetVerbatimFormat(reply);
+}
+
+/* Return the format value of an verbatim string reply. */
+const char* RM_CallReplyVerbatimString(RedisModuleCallReply *reply, size_t* len) {
+    return callReplyGetVerbatimString(reply, len);
+}
+
+/* Return the boolean value of an boolean reply. */
+int RM_CallReplyBool(RedisModuleCallReply *reply) {
+    return callReplyGetBool(reply);
+}
+
+/* Return the 'idx'-th nested call reply element of a set reply, or NULL
+ * if the reply type is wrong or the index is out of range. */
+RedisModuleCallReply *RM_CallReplySetElement(RedisModuleCallReply *reply, size_t idx) {
+    return callReplyGetSetElement(reply, idx);
+}
+
+/* Return the 'idx'-th key of a map reply, or NULL
+ * if the reply type is wrong or the index is out of range. */
+RedisModuleCallReply *RM_CallReplyMapKey(RedisModuleCallReply *reply, size_t idx) {
+    return callReplyGetMapKey(reply, idx);
+}
+
+/* Return the 'idx'-th value of a map reply, or NULL
+ * if the reply type is wrong or the index is out of range. */
+RedisModuleCallReply *RM_CallReplyMapVal(RedisModuleCallReply *reply, size_t idx) {
+    return callReplyGetMapVal(reply, idx);
+}
+
+/* Return the 'idx'-th key of a attribute reply, or NULL
+ * if the reply type is wrong or the index is out of range. */
+RedisModuleCallReply *RM_CallReplyAttributeKey(RedisModuleCallReply *reply, size_t idx) {
+    return callReplyGetAttributeKey(reply, idx);
+}
+
+/* Return the 'idx'-th value of a attribute reply, or NULL
+ * if the reply type is wrong or the index is out of range. */
+RedisModuleCallReply *RM_CallReplyAttributeVal(RedisModuleCallReply *reply, size_t idx) {
+    return callReplyGetAttributeVal(reply, idx);
+}
+
+/* Return the attribute of the given reply, or NULL if no attribute exists. */
+RedisModuleCallReply *RM_CallReplyAttribute(RedisModuleCallReply *reply) {
+    return callReplyGetAttribute(reply);
 }
 
 /* Return the pointer and length of a string or error reply. */
 const char *RM_CallReplyStringPtr(RedisModuleCallReply *reply, size_t *len) {
-    moduleParseCallReply(reply);
-    if (reply->type != REDISMODULE_REPLY_STRING &&
-        reply->type != REDISMODULE_REPLY_ERROR) return NULL;
-    if (len) *len = reply->len;
-    return reply->val.str;
+    size_t private_len;
+    if (!len) {
+        len = &private_len;
+    }
```

Comment:
i'd rather not waste so many lines on that nonsense.
```suggestion
    if (!len) len = &private_len;
```